### PR TITLE
fix(manager): only reset custom gas price

### PR DESCRIPTION
**Pull request by tuxcanfly**
## Overview

This PR fixes custom gas price reset after a successful submission.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

## Summary by CodeRabbit

- **Bug Fixes**
	- Modified the logic for adjusting `gasPrice` to prevent division operation when not required.
- **Refactor**
	- Refactored test cases for submitting blocks to a mock DA to cover different scenarios.
- **Validation**
	- Added a validation check for `DAGasMultiplier` in `nodeConfig` to meet specific criteria.

### CODE REVIEW

**Review by gupadhyaya — DISMISSED** [review]

**Review by Manav-Aggarwal — DISMISSED** [review]

**Comment by tuxcanfly** [comment]
Adding a few more sanity checks while here.

**Review by MSevey — CHANGES_REQUESTED** [review]
some minor test clean up I think.

File: block/manager_test.go [hunk 1]
```diff
@@ -138,49 +138,74 @@ func TestIsDAIncluded(t *testing.T) {
...
-		require.NoError(t, err)
-		err = m.submitBlocksToDA(ctx)
-		require.NoError(t, err)
-		mockDA.AssertExpectations(t)
-	})
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			mockDA := &mock.MockDA{}
+			m := getManager(t, mockDA)
+			m.logger = test.NewLogger(t)
```
[the remark this answers is not in the document]
Comment (tuxcanfly) [reply block/manager_test.go:166]: Stray change, removed.

File: block/manager_test.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ import (
 
 // Returns a minimalistic block manager
 func getManager(t *testing.T, backend goDA.DA) *Manager {
-	logger := test.NewFileLoggerCustom(t, test.TempLogFileName(t, t.Name()))
+	logger := test.NewLogger(t)
```
[the remark this answers is not in the document]
Comment (tuxcanfly) [reply block/manager_test.go:31]: There seems to be an issue with the logger if the name contains a path separator:

```
    --- FAIL: TestSubmitBlocksToMockDA/defaults (0.00s)
panic: createtemp TestSubmitBlocksToMockDA_defaultsTestSubmitBlocksToMockDA/defaults.log: pattern contains path separator [recovered]
        panic: createtemp TestSubmitBlocksToMockDA_defaultsTestSubmitBlocksToMockDA/defaults.log: pattern contains path separator
```

Besides, the test output is much more easier to debug with the raw log in stdout:

https://gist.github.com/tuxcanfly/dd0d07ca8c785d2ea80c088a8120a8e5

**Review by MSevey — APPROVED** [review]

File: cmd/rollkit/docs/rollkit_start.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ rollkit start [flags]
       --rollkit.da_address string                       DA address (host:port) (default "http://localhost:26658")
       --rollkit.da_auth_token string                    DA auth token
       --rollkit.da_block_time duration                  DA chain block time (for syncing) (default 15s)
-      --rollkit.da_gas_multiplier float                 DA gas price multiplier for retrying blob transactions (default -1)
+      --rollkit.da_gas_multiplier float                 DA gas price multiplier for retrying blob transactions
```
Comment (gupadhyaya) [inline cmd/rollkit/docs/rollkit_start.md:36]: do you still want to say `default 0`?

**Review by gupadhyaya — APPROVED** [review]

File: cmd/rollkit/docs/rollkit_start.md (hunk 3 above)
Comment (tuxcanfly) [reply cmd/rollkit/docs/rollkit_start.md:36]: It seems to be autogenerated. I think it's fine.

**Pull request merged**